indico/modules/core/captcha.py

```diff
@@ -0,0 +1,48 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2022 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from flask import session
+from marshmallow import ValidationError, fields
+
+from indico.core.plugins import plugin_engine
+from indico.modules.events.registration.plugins import CaptchaPluginMixin
+from indico.util.i18n import _
+
+
+class CaptchaField(fields.String):
+    """Validates an answer to a CAPTCHA."""
+
+    def __init__(self):
+        def validate(user_answer):
+            if plugin := get_captcha_plugin():
+                if not plugin.validate_captcha(user_answer):
+                    raise ValidationError(_('Incorrect answer'))
+            else:
+                answer = session.get('captcha_state')
+                if not answer or answer != user_answer:
+                    raise ValidationError(_('Incorrect answer'))
+
+        super().__init__(required=True, validate=validate)
```

Comment:
I think you are supposed to raise those errors from within `_deserialize`: https://marshmallow.readthedocs.io/en/stable/custom_fields.html#creating-a-field-class